plugins/kotlin/frontend-independent/src/org/jetbrains/kotlin/idea/util/psiUtils.kt

```diff
@@ -135,3 +137,56 @@ tailrec fun KtCallableDeclaration.isExplicitTypeReferenceNeededForTypeInference(
         else -> false
     }
 }
+
+fun KtPropertyAccessor.isRedundantSetter(): Boolean {
+    if (!isSetter) return false
+    val expression = bodyExpression ?: return canBeCompletelyDeleted()
+    if (expression is KtBlockExpression) {
+        val statement = expression.statements.singleOrNull() ?: return false
+        val parameter = valueParameters.singleOrNull() ?: return false
+        val binaryExpression = statement as? KtBinaryExpression ?: return false
+        return binaryExpression.operationToken == KtTokens.EQ
+                && binaryExpression.left?.isBackingFieldReferenceTo(property) == true
+                && binaryExpression.right?.mainReference?.resolve() == parameter
+    }
+    return false
+}
+
+fun removeRedundantSetter(setter: KtPropertyAccessor) {
```

Comment:
Nit: what about moving `removeRedundantGetter` too?

Reply:
Ah, yeah, why not?